sycl/include/CL/sycl/detail/context_impl.hpp

```diff
@@ -138,6 +138,9 @@ class context_impl {
 
   KernelProgramCache &getKernelProgramCache() const;
 
+  /// Returns true if and only if context contains the given device.
+  bool hasDevice(std::shared_ptr<detail::device_impl> Device) const;
```

Comment:
@alexbatashev, is it okay to use `std::shared_ptr` or we should switch to `shared_ptr_class`?

Reply:
In this particular case we should switch to `shared_ptr_class`. If I'm not mistaken, there are no more places in the code, that store or accept `device_impl` as `std::shared_ptr`.

Reply:
sycl/include/CL/sycl/detail/queue_impl.hpp:

`using DeviceImplPtr = std::shared_ptr<detail::device_impl>;`

Reply:
Ok, I did that additional fix for device_impl.